### DISCUSSION TITLE
Firebase FCM getToken() throws apns-token-not-set error

### DISCUSSION THREAD

**Discussion by stevenc1981** [firebase/flutterfire#12815] [Q&A]
This seems like a bug to me but wants some opinion from the experts.

On some iOS devices, user is getting this APNS token unavailable error when executing `FirebaseMessaging.instance.getToken()`:

      [firebase_messaging/apns-token-not-set] APNS token has not been set yet. Please ensure the APNS token is available by calling `getAPNSToken()`.

      Fatal Exception: FlutterError
      0  App                            0x478967 MethodChannelFirebaseMessaging._APNSTokenCheck + 138 (method_channel_messaging.dart:138)
      1  App                            0x4787d3 MethodChannelFirebaseMessaging.getToken + 244 (method_channel_messaging.dart:244)
      2  App                            0x40c9c0 main.<anonymous closure> (#5) + 92 (main.dart:92)

Based on what I can see in the flutterfire code: https://github.com/firebase/flutterfire/blob/master/packages/firebase_messaging/firebase_messaging_platform_interface/lib/src/method_channel/method_channel_messaging.dart

It is calling `_APNSTokenCheck()` which calls `getAPNSToken()` anyways. Sure, I can call `getAPNSToken()` before `getToken()` but there isn't any difference. I'll get null returned anyways. And even if I do a null check with `getAPNSToken()`, Android devices will get null from `getAPNSToken()` and `getToken()` will not get executed.

I think `getToken()` should handle both iOS and Android scenarios gracefully instead of throwing errors.

**Comment by alefwd** [comment]
I see this exception daily using Flutter, and I can't understand why this is happening, nothing mentioned in the docs :/

**Comment by jeet-appcrunk** [comment]
any solution i am getting null in getAPNToken()

**Comment by EmadBeltaje** [comment]
any updates? :')
